src/Tribe/Body_Classes.php

```diff
@@ -0,0 +1,367 @@
+	 *
+	 * @since TBD
+	 *
+	 * @param string $class The class we are checking for.
+	 * @param string $queue The queue we want to check 'admin', 'display', 'all'
+	 * @return boolean
+	 */
+	public function class_is_enqueued( $class, $queue = 'display' ) {
+		$classes = $this->get_classes_for_queue( $queue );
+		if ( ! $this->class_exists( $class ) ) {
+			return false;
+		}
+
+		return $classes[ $class ];
+	}
+
+	/**
+	 * Dequeues a class.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $class
+	 * @param string $queue The queue we want to alter 'admin', 'display', 'all'
+	 * @return void|false
+	 */
+	public function dequeue_class( $class, $queue = 'display' ) {
+		$classes = $this->get_classes_for_queue( $queue );
+		if ( ! $this->class_exists( $class ) ) {
+			return false;
+		}
+
+		$classes[ $class ] = false;
+	}
+
+	/**
+	 * Enqueues a class.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $class
+	 * @param string $queue The queue we want to alter 'admin', 'display', 'all'
+	 * @return void|false
+	 */
+	public function enqueue_class( $class, $queue = 'display' ) {
+		$classes = $this->get_classes_for_queue( $queue );
+		if ( ! $this->class_exists( $class ) ) {
+			return false;
+		}
+
+		$classes[ $class ] = true;
+	}
+
+	/**
+	 * Add a single class to the queue.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $class The class to add.
+	 * @param string $queue The queue we want to alter 'admin', 'display', 'all'
+	 * @return void
+	 */
+	public function add_class( $class, $queue = 'display' ) {
+		if ( empty( $class ) ) {
+			return;
+		}
+
+		if ( is_array( $class ) ) {
+			$this->add_classes( $class, $queue );
+		} elseif ( $this->should_add_body_class_to_queue( $class, $queue ) ) {
+
+			$class   = sanitize_html_class( $class );
+
+			if ( 'admin' !== $queue ) {
+				$classes = $this->get_classes_for_queue();
+				$classes[ $class ] = true ;
+			}
+
+			if ( 'display' !== $queue ) {
+				$classes = $this->get_classes_for_queue( 'admin' );
+				$classes[ $class ] = true ;
+			}
+
+		}
+	}
+
+	/**
+	 * Add an array of classes to the queue.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string> $class The classes to add.
+	 * @return void
+	 */
+	public function add_classes( array $classes, $queue = 'display' ) {
+		$classes = array_map( 'sanitize_html_class', $classes );
+
+		foreach ( $classes as $key => $value ) {
+			// If the classes are passed as class => bool, only add ones set to true.
+			if ( ! is_string( $value ) && false !== $value  ) {
+				$this->add_class( $key, $queue );
+			} else {
+				$this->add_class( $value, $queue );
+			}
+		}
+	}
+
+	/**
+	 * Remove a single class from the queue.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $class The class to remove.
+	 * @return void
+	 */
+	public function remove_class( $class, $queue = 'display' ) {
+		$classes = $this->get_classes_for_queue( $queue );
+
+		if ( 'admin' !== $queue ) {
+			$this->classes = array_filter(
+				$this->classes,
+				function( $k ) use ( $class ) {
```

Comment:
I think you can do this, but lets also test.
```suggestion
				static function( $k ) use ( $class ) {
```